gift_card.py

```diff
@@ -348,3 +352,129 @@ def parse(cls, report, records, data, localcontext):
+
+    done_msg = fields.Text("Message", readonly=True)
+
+    @staticmethod
+    def default_done_msg():  # pragma: no cover
+        """
+        Returns default message.
+        """
+        return (
+            'Gift card redemption is complete.'
+        )
+
+
+class GiftCardRedeemWizard(Wizard):
+    "Gift Card Redeem Wizard"
+    __name__ = 'gift_card.redeem.wizard'
+
+    start = StateView(
+        'gift_card.redeem.start',
+        'gift_card.redeem_start_view_form',
+        [
+            Button('Cancel', 'end', 'tryton-cancel'),
+            Button(
+                'Save', 'do',
+                'tryton-ok', default=True
+            )
+        ]
+    )
+    do = StateTransition()
+    done = StateView(
+        'gift_card.redeem.end',
+        'gift_card.redeem_done_view_form',
+        [
+            Button('OK', 'end', 'tryton-ok')
+        ]
+    )
+
+    @classmethod
+    def __setup__(cls):
+        super(GiftCardRedeemWizard, cls).__setup__()
+        cls._error_messages.update({
+            'gift_card_inactive': (
+                'The gift card to be redeemed must be in active state.'
+            ),
+            'gift_card_redeemed': (
+                'This gift card has already been redeemed once.'
+            ),
+        })
+
+    def default_start(self, data):
+        """
+        Initial state of redeem wizard.
+        """
+        Gateway = Pool().get('payment_gateway.gateway')
+        GiftCard = Pool().get('gift_card.gift_card')
+
+        gift_card = GiftCard(Transaction().context.get('active_id'))
+        self.check_giftcard_state(gift_card)
+
+        gateways = Gateway.search([
+            ('method', '=', 'gift_card'),
+        ])
+        return {
+            'description': '',
+            'gateway': gateways and gateways[0].id or None,
+        }
+
+    def transition_do(self):
+        """
+        Transition where PaymentTransaction is created and associated
+        with current gift card.
+        """
+        GiftCard = Pool().get('gift_card.gift_card')
+        PaymentTransaction = Pool().get('payment_gateway.transaction')
+        Date = Pool().get('ir.date')
+
+        gift_card = GiftCard(Transaction().context.get('active_id'))
+
+        transaction, = PaymentTransaction.create([{
+            'description': self.start.description or 'Redeeming gift card',
+            'date': Date.today(),
+            'party': self.start.party.id,
+            'address': (
+                self.start.party.addresses and
+                self.start.party.addresses[0].id or None
+            ),
+            'amount': gift_card.amount,
+            'currency': gift_card.currency.id,
+            'gateway': self.start.gateway.id,
+            'gift_card': gift_card.id,
+        }])
+        GiftCard.write([gift_card], {
+            'payment_transactions': [('add', [transaction.id])],
+        })
+        transaction.capture_self()
+        transaction.post(gift_card.payment_transactions)
```

Comment:
Just capture. It must automatically post https://github.com/openlabs/payment-gateway/blob/develop/manual.py#L70